docs/multi_credential_assignment.md

```diff
@@ -220,3 +220,16 @@ POST /api/v2/job_templates/N/launch/
     }
 }
 ```
+
+Inventory Source Credentials
+----------------------------
+
+Inventory sources and inventory updates that they spawn also use the same
+relationship. The new endpoints for this are
+ - `/api/v2/inventory_sources/N/credentials/` and
+ - `/api/v2/inventory_updates/N/credentials/`
+
+Most cloud sources will continue to adhere to the constraint that they
+must have a single credential that corresponds to their cloud type.
+However, this relationship allows users to associate multiple vault
```

Comment:
One important question we maybe didn't answer succinctly in our meeting given:

> However, this relationship allows users to associate multiple vault credentials of different ids to inventory sources.

Are we comfortable shrugging our shoulders about what this _means_ until we actually implement it later?  If somebody actually uses this API to assign multi vault, what does it _do_?  What _should_ it do?  Are we comfortable with ambiguity?

Reply:
Discussion is in #223, and you go down one of two roads.

Either you say that (1) it's okay to import vaulted content into our current system, so it goes into group / host variables plaintext, or (2) this content needs to be redacted / encrypted like survey passwords.

(1) is re-implementing the prior PR on top of this... probably a day to have a PR due to the need to switch to pexpect.

Option (2) won't be in the cards for a _long_ time. Inventory imports run through `ansible-inventory`, and no mechanism exists to know which values were from vault. So we're talking about a new feature request for Ansible core, then adding the flag to use it, developing a solution for old versions _without_ the feature. We could say that the presence of _any_ vault credentials means that all variables need to be saved encrypted, which is... yuck.

I would like to give a little push toward option (1). Firstly, consider the quite substantial community demand expressed. One comment in that discussion got _15_ 👍, and I think this kind of response should factor into our plan to at least some degree. AFAIK, nothing about the community feature request necessitates encryption. Option (1) also could latter be expanded to (2), in a future world where the pre-reqs are available and users have asked for it. If this gets into the UI (which I also realize is a discussion), then documentation needs to be self-evident end-to-end so that users know they will _see_ variables right away after import.

Reply:
Understood.  What I'm asking is do we want to merge _this_ PR without solutions #1 and #2 existing?  Are we comfortable with the API supporting vault assignment in an ambiguous way until there's an actual solution for #223?

Reply:
I hear you there. For this PR, I could prohibit vault creds and put in a TODO about that so we don't have any transient inconsistent behavior. That would help to keep these different tasks separate. This still feels right to me as 2 separate PRs.

I've been hitting up @jakemcdermott and @jladdjr about the possible feature drop.

Reply:
Yea, I agree that that the actual implementation makes sense in a separate PR - this PR is good as is, but perhaps we could block Vault for now.

Reply:
Right, this'll get dropped back in as part of that.

```python
# Vault credentials are only exclusive with others of same ID
if sub.unique_hash() in [cred.unique_hash() for cred in parent.credentials.all()]:
    return {"msg": _(
        "A credential of type {credential_type} is already assigned to this inventory source."
    ).format(credential_type=sub.unique_hash(display=True))}
```